src/format.py

```diff
@@ -9,33 +9,38 @@
 from __future__ import print_function
 
 import curses
-
 import parse
-
+import copy
+from formattedText import FormattedText
 
 class SimpleLine(object):
 
-    def __init__(self, line, index):
-        self.line = line
+    def __init__(self, formattedLine, index):
+        self.formattedLine = formattedLine
         self.index = index
 
     def printOut(self):
         print(str(self))
 
-    def output(self, stdscr):
+    def output(self, printer):
         (minx, miny, maxx, maxy) = self.controller.getChromeBoundaries()
-        maxLen = maxx - minx
+        maxLen = min(maxx - minx, len(str(self)))
         y = miny + self.index + self.controller.getScrollOffset()
+
+        if (y < miny or y > maxy):
+            # wont be displayed!
+            return
+
         try:
-            stdscr.addstr(y, minx, str(self)[0:maxLen])
-        except curses.error:
+            self.formattedLine.printText(y, minx, printer, maxLen)
+        except curses.error as e:
```

Comment:
Yeah, so interesting thing: I was getting the exception before because `SimpleLine` didn't have the guard around y height being too big or too small. In current `master` we are actually throwing and catching QUITE A LOT - which might have some perf ramnifications on super big files.

After adding the guard check I didn't get any more curses exceptions. Personally I think we should allow those to propogate out and if you agree I will remove on both prints